core/src/test/java/org/bitcoinj/script/ScriptPatternTest.java

```diff
@@ -92,4 +97,19 @@ public void testCreateOpReturnScript() {
                 ScriptBuilder.createOpReturnScript(new byte[10])
         ));
     }
+
+    @Test
+    public void p2shScriptHashFromKeys() {
+        // import some keys from this example: https://gist.github.com/gavinandresen/3966071
+        ECKey key1 = DumpedPrivateKey.fromBase58(MAINNET, "5JaTXbAUmfPYZFRwrYaALK48fN6sFJp4rHqq2QSXs8ucfpE4yQU").getKey();
+        key1 = ECKey.fromPrivate(key1.getPrivKeyBytes());
+        ECKey key2 = DumpedPrivateKey.fromBase58(MAINNET, "5Jb7fCeh1Wtm4yBBg3q3XbT6B525i17kVhy3vMC9AqfR6FH2qGk").getKey();
+        key2 = ECKey.fromPrivate(key2.getPrivKeyBytes());
+        ECKey key3 = DumpedPrivateKey.fromBase58(MAINNET, "5JFjmGo5Fww9p8gvx48qBYDJNAzR9pmH5S389axMtDyPT8ddqmw").getKey();
+        key3 = ECKey.fromPrivate(key3.getPrivKeyBytes());
```

Comment:
Why is each key initialized twice?

Reply:
No idea. I didn't look at the actual code (-:

Reply:
We can fix that in a future commit.